### PR TITLE
Document Dependencies

### LINKED ISSUE #3047: Explicitly list our dependencies
**Describe the current behavior/improvement**

As identified by @super-harsh in PR #3041, it would be helpful for users who don't want to rely on `install-dependencies.sh` to have a canonical list of things they should install manually in order to do ASO development.

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Adds documentation of the developer dependencies required to work on ASO if you don't want to use our devcontainer or run `install-dependencies.sh`.

Closes #3047 

**Special notes for your reviewer**:

Documentation is generated by scanning `install-dependencies.sh` to help ensure that it remains up to date.

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains documentation

### CODE REVIEW

File: .devcontainer/install-dependencies.sh [hunk 1]
```diff
@@ -24,6 +24,17 @@ set -eu
 # -s --skip-installed   : Skip anything that's already installed
 #
 
+#
+# This file includes documentation in lines prefixed `#doc#`.
+# These lines are extracted by running `task doc:dependencies` from the root folder.
+#
+# Each depencency should be identified by a single line with the following format:
+#
+# | <name> | <version> | <details> |
```
Comment (matthchr) [inline .devcontainer/install-dependencies.sh:33]: Missing the `#doc` prefix?

File: .devcontainer/install-dependencies.sh [hunk 2]
```diff
@@ -82,6 +93,7 @@ fi
 
 # Ensure we have the right version of GO
 
+## | Go | 1.20 | https://golang.org/doc/install #
```
Comment (matthchr) [inline .devcontainer/install-dependencies.sh:96]: Missing  the doc prefix

File: .devcontainer/install-dependencies.sh [hunk 3]
```diff
@@ -99,6 +111,7 @@ fi
 
 # Ensure we have AZ
 
+## | AZ | latest | https://docs.microsoft.com/en-us/cli/azure/install-azure-cli |
```
Comment (matthchr) [inline .devcontainer/install-dependencies.sh:114]: Missing as well (kudos to @super-harsh for actually noticing this)

File: Taskfile.yml [hunk 4]
```diff
@@ -1024,6 +1024,13 @@ tasks:
     cmds:
       - "{{.SCRIPTS_ROOT}}/check_frontmatter.py"
 
+  doc:dependencies:
+    desc: Generate documentation for depencencies
+    cmds:
+      - cp ./docs/v2/dependencies-header.md ./docs/hugo/content/contributing/dependencies.md
```
Comment (matthchr) [inline Taskfile.yml:1030]: Does `cp` overwrite the file with no prompting if it already exists? I guess yes unless you use `-i`?

**Review by matthchr — APPROVED** [review]

File: .devcontainer/install-dependencies.sh (hunk 1 above)
Comment (theunrepentantgeek) [reply .devcontainer/install-dependencies.sh:33]: No, this is documentation on how to use it - the table header is in another file.

File: .devcontainer/install-dependencies.sh (hunk 2 above)
Comment (theunrepentantgeek) [reply .devcontainer/install-dependencies.sh:96]: Fixed.

File: .devcontainer/install-dependencies.sh (hunk 3 above)
Comment (theunrepentantgeek) [reply .devcontainer/install-dependencies.sh:114]: Fixed.

File: Taskfile.yml (hunk 4 above)
Comment (theunrepentantgeek) [reply Taskfile.yml:1030]: Yes.

**Pull request merged**